src/sss_client/pam_sss.c

```diff
@@ -594,20 +594,26 @@ static int user_info_expire_warn(pam_handle_t *pamh,
         return PAM_BUF_ERR;
     }
     memcpy(&expire, buf + sizeof(uint32_t), sizeof(uint32_t));
-    if (expire >= DAYSEC) {
-        expire /= DAYSEC;
-        unit = _("day(s)");
-    } else if (expire >= HOURSEC) {
-        expire /= HOURSEC;
-        unit = _("hour(s)");
-    } else if (expire >= MINSEC) {
-        expire /= MINSEC;
-        unit = _("minute(s)");
+    /* expire == 0 indicates the password expired */
+    if (expire != 0) {
+        if (expire >= DAYSEC) {
+            expire /= DAYSEC;
+            unit = _("day(s)");
+        } else if (expire >= HOURSEC) {
```

Comment:
The same comment as about ngettext & plural

Reply:
Fixed.